hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/preloader/NearCachePreloader.java

```diff
@@ -234,9 +223,9 @@ private int loadKeySet(BufferingInputStream bis, DataStructureAdapter<Data, ?> a
         return loadedKeys;
     }
 
-    private void writeKeySet(FileOutputStream fos, FileChannel outChannel, Set<KS> keySet) throws IOException {
-        for (KS key : keySet) {
-            Data dataKey = serializationService.toData(key);
+    private void writeKeySet(FileOutputStream fos, FileChannel outChannel, Iterator keySetIterator) throws IOException {
```

Comment:
It's an iterator over the keySet of a data structure.
```
map.keySet().iterator()
map.values().iterator()
map.entrySet().iterator()
```
We want the `keySet().iterator()` here, therefore the name.